deep_speech_2/decoder.py

```diff
@@ -36,24 +37,197 @@ def ctc_best_path_decode(probs_seq, vocabulary):
     return ''.join([vocabulary[index] for index in index_list])
 
 
-def ctc_decode(probs_seq, vocabulary, method):
-    """CTC-like sequence decoding from a sequence of likelihood probablilites.
+def ctc_beam_search_decoder(probs_seq,
+                            beam_size,
+                            vocabulary,
+                            blank_id=0,
+                            cutoff_prob=1.0,
+                            ext_scoring_func=None,
+                            nproc=False):
```

Comment:
Preserved temporarily before fixing the problem about how to pass ext_scoring_fuc to the multi processes.